needy/cd.py

```diff
@@ -2,9 +2,14 @@
 
 _current_directory = None
 
+
 def current_directory():
     """ returns the current directory as given to cd (which means unresolved symlinks) """
     global _current_directory
+
+    if os.path.realpath(os.environ['PWD']) is not os.path.realpath(os.getcwd()):
```

Comment:
this should probably go a bit further down since `_current_directory` takes priority over PWD. i.e. replace the `return`s with `ret = `, then use `ret` instead of `os.environ['PWD']` for this check